apis/composition.yaml

```diff
@@ -0,0 +1,277 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: xnetworks.aws.platform.upbound.io
+  labels:
+    provider: aws
+    type: basic
+spec:
+  compositeTypeRef:
+    apiVersion: aws.platform.upbound.io/v1alpha1
+    kind: XNetwork
+  mode: Pipeline
+  pipeline:
+    - step: render-templates
+      functionRef:
+        name: crossplane-contrib-function-go-templating
+      input:
+        apiVersion: gotemplating.fn.crossplane.io/v1beta1
+        kind: GoTemplate
+        source: Inline
+        inline:
+          template: |
+            {{ $params := .observed.composite.resource.spec.parameters }}
+
+            {{- define "providerConfigRef" }}
+              providerConfigRef:
+                name: {{ .observed.composite.resource.spec.parameters.providerConfigName  }}
+            {{- end }}
+            {{- define "networkId" }}
+              networks.aws.platform.upbound.io/network-id: {{ .observed.composite.resource.spec.parameters.id }}
+            {{- end }}
+            {{- define "region" }}
+              region: {{ .observed.composite.resource.spec.parameters.region }}
+            {{- end }}
+            {{- define "deletionPolicy" }}
+              deletionPolicy: {{ .observed.composite.resource.spec.parameters.deletionPolicy | default "Delete" }}
+            {{- end }}
```

Comment:
```suggestion
```

I suggest to remove usage of `define` and `include` and use consistent straightforward

```
              providerConfigRef:
                name: {{ $params.providerConfigName }}
              deletionPolicy: {{ $deletionPolicy }}
```
for every composed resource, similar to what you already did for Subnets below
https://github.com/upbound/configuration-aws-network/pull/56/files#diff-301e9525d55cf383a2c67f73a0c8e203053a486fc61ba64983b47a070cc358edR110-R112